src/Action-Repository.js

```diff
@@ -0,0 +1,19 @@
+class ActionRepository {
+  constructor(data) {
+    this.data = data;
+  }
+  findCurrentUserData(userId) {
+    return this.data.filter((activityObj) => activityObj.userID === userId);
+  }
+  returnActionByDate(userId, date, action) {
+    return this.findCurrentUserData(userId).find(user => {
+      return user.date === date;
+    })[action];
+  }
+  returnActionByWeek(userId, date, action) {
+    let index = this.findCurrentUserData(userId).findIndex((actionObj) => actionObj.date === date);
+    return this.findCurrentUserData(userId).map(actionObj => actionObj[action]).splice(index - 6, 7);
+  }
```

Comment:
All of these abstractions of functionality make sense -- just make sure you explicitly test that things will work for each dataset, especially since not every set requires an `action` argument to get passed in